lib/app/admin/data-reset/create.js

```diff
@@ -0,0 +1,119 @@
+var q = require('q');
+var async = require('async');
+var relationsMapper = require('./map_relations');
+
+var DEFAULT_LIST_RELATIONS = require('./default_list_relations');
+var DEFAULT_ITEM_LIST = require('./default_item_type_list');
+
+
+function getBuildDataFunction(mediator) {
+  return function buildData() {
+    var deferred = q.defer();
+    var relationsMap = [];
+    console.log('>>>>>Start create');
+    async.eachSeries(DEFAULT_ITEM_LIST, function(itemType, eachTypeCb) {
+      createItemsForType(mediator, itemType, relationsMap, function createItemsCb(err, mappingsForType) {
+        relationsMap[itemType] = mappingsForType;
+        eachTypeCb();
+      });
+    }, function(err) {
+      console.log('>>>>>create end');
+      (err) ? deferred.reject(err) : deferred.resolve();
+    });
+
+    return deferred.promise;
+  };
+}
+/**
+ * Builds data-reset using seed data-reset located in ./data-reset-demo, synchronously publishes 'wfm:cloud:{itemType}:create topic for each item.
+ * @param {Object} mediator  - mediator object used for communication
+ * @returns {Promise} Promise, resolved if all items are successfully created.
+ */
+// var buildDataDo = function(mediator) {
+//   var deferred = q.defer();
+//   var relationsMap = [];
+//   console.log('>>>>>Start create');
+//   async.eachSeries(DEFAULT_ITEM_LIST, function(itemType, eachTypeCb) {
+//     createItemsForType(mediator, itemType, relationsMap, function createItemsCb(err, mappingsForType) {
+//       relationsMap[itemType] = mappingsForType;
+//       eachTypeCb();
+//     });
+//   }, function(err) {
+//     console.log('>>>>>create end');
+//     (err) ? deferred.reject(err) : deferred.resolve();
+//   });
+//
+//   return deferred.promise;
+// };
+
+
+/**
+ * @callback createItemsCb
+ * @param {Error} err - an error or null
+ * @param {Array} mappingsForType - mapping between seed data-reset id and newly created dada ids
+ */
+
+/**
+ * Creates all items of a given type
+ * @param {Object} mediator  - mediator object used for communication
+ * @param {String} itemType - type of the item to remove e.g. result, workorder
+ * @param {Array} relationsMap - An array mapping out relations between seed and newly created data-reset
+ *
+ * @param {createItemsCb} createItemsCb - callback called after all items for type have been created
+ */
+var createItemsForType = function(mediator, itemType, relationsMap, createItemsCb) {
```

Comment:
I don't understand why the first function in this file returns a promise, but the other two are callbacks? The API should be consistent.